components/bastion/setup.ftl

```diff
@@ -26,41 +26,61 @@
   [#local networkVnetResource = networkResources["vnet"]]
   [#local mgmtSubnetResource = networkResources["subnets"]["mgmt"]["subnet"]]
```

Comment:
The bastion doesn't have to be in the mgmt subnet, it can be in any of them. We should have some routines for getting the subnet based on the tier the component is in

Reply:
Have added this in 2893fda - a new function had to be created as Azure doesn't implement zones in network component, so `getSubnets()` from the shared provider doesn't quite work out.

It adds a little bit of clarity over the process of just drilling down from the networkResources every time, so I think its worthwhile.